src/containers/SearchBar.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { fetchWeather } from '../actions/index';
+
+
+class SearchBar extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {term : ''};
+        this.onInputChange = this.onInputChange.bind(this);
+        this.onSubmit = this.onSubmit.bind(this);
+    }
+
+    onInputChange(event) {
```

Comment:
rename to `handleChange` 
https://reactjs.org/docs/forms.html#controlled-components